internal/kube/client.go

```diff
@@ -356,6 +356,7 @@ func (c *client) DefaultNamespace() string {
 }
 
 // watch provides a unified implementation to watch resources.
+// timeout of zero does NOT mean "no timeout", i.e. "wait forever" - it means the function will time-out almost immediately - at the discretion of the golang scheduler.
 func watch(res string, watchFn func() (bool, error), timeout time.Duration) error {
```

Comment:
I think this 0 equals immediately timeout is against principle of least surprise, normally for timeout, either have a time period or disable timeout. So here I actually expected if 0 means no timeout, or have a validation check here to not accept 0.

Reply:
Yes, I am aware of that problem. But changing the contract now may require more work. How about that:
- We merge the PR so that the pipelines are not failing.
- I create an issue for changing the contract on this method and we estimate it's priority

WDYT?

Reply:
sure, just mentioned in that followup issue about this PR.